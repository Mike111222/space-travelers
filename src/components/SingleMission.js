import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import {
  joinedMission,
  leavedMission,
} from '../redux/mission/missionSlice';

const SingleMission = ({ mission }) => {
  const dispatch = useDispatch();
  return (
    <tr>
      <td>{mission.missionName}</td>
      <td>{mission.description}</td>
      <td>
        <div
          className={mission?.joined ? 'member-badge active-mission' : 'member-badge'}
        >
          {mission?.joined ? 'Active Member' : 'NOT A MEMBER'}
        </div>
      </td>
      <td>
        <button
          className={mission.joined ? 'leaveBtn' : 'joinBtn'}
          type="button"
          onClick={() => {
            if (mission.joined) {
              dispatch(leavedMission(mission.id));
            } else {
              dispatch(joinedMission(mission.id));
            }
          }}
        >
          {mission?.joined ? 'Leave Mission' : 'Join Mission'}
        </button>
      </td>
    </tr>
  );
};

SingleMission.propTypes = {
  mission: PropTypes.shape({
    id: PropTypes.string.isRequired,
    missionName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    joined: PropTypes.bool,
  }).isRequired,
};
export default SingleMission;
