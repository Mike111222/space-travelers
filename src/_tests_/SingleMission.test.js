import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import SingleMission from '../components/SingleMission';

const mockStore = configureStore([]);

describe('SingleMission', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it('should match the snapshot', () => {
    const mission = {
      id: '1',
      missionName: 'Mission 1',
      description: 'Mission 1 description',
      joined: false,
    };

    const { asFragment } = render(
      <Provider store={store}>
        <SingleMission mission={mission} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
