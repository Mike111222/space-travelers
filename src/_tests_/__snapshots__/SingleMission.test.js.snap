// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMission should match the snapshot 1`] = `
<DocumentFragment>
  Mission 1Mission 1 description
  <div
    class="member-badge"
  >
    NOT A MEMBER
  </div>
  <button
    class="joinBtn"
    type="button"
  >
    Join Mission
  </button>
</DocumentFragment>
`;
